jubilant/_juju.py

```diff
@@ -290,6 +290,32 @@ def destroy_model(
         if model == self.model:
             self.model = None
 
+    def integrate(self, app1: str, app2: str, *, via: str | Iterable[str] | None = None) -> None:
```

Comment:
How would you feel about supporting actual IPvXNetwork objects, like:

```suggestion
    def integrate(self, app1: str, app2: str, *, via: str | ipaddress.IPv4Network | ipaddress.IPv6Network | Iterable[str | ipaddress.IPv4Network | ipaddress.IPv6Network ] | None = None) -> None:
```

(You could have a private alias to make that shorter and more readable).

It should only require changing the via to be:

```python
        if via:
            if isinstance(via, (str, ipaddress.IPv4Network, ipaddress.IPv6Network)):
                args.extend(['--via', str(via)])
            else:
                args.extend(['--via', ','.join(str(v) for v in via))])
```

Reply:
It's a reasonable suggestion. However, I'm inclined to keep them simple types for v1 at least. I'm skeptical that `IPvXNetwork` wouldn't be used instead of just `str`, in this context at any rate, and it doesn't seem worth adding type complexity for until we see that it is desired.